src/mono/mono/mini/monovm.c

```diff
@@ -226,6 +226,40 @@ monovm_initialize (int propertyCount, const char **propertyKeys, const char **pr
 	return 0;
 }
 
+// Initialize monovm with properties set by runtimeconfig.json. Primarily used by mobile targets.
+int
+monovm_runtimeconfig_initialize (MonovmRuntimeConfigArguments *arg, MonovmRuntimeConfigArgumentsCleanup cleanup_fn, void *user_data)
+{
+	switch (arg->kind) {
+	case 0: {
+		MonoFileMap *map;
+		void *ret_handle;
+		char *buffer;
+
+		map = mono_file_map_open (arg->runtimeconfig.name.path);
+		g_assert (map);
+		arg->runtimeconfig.data.data_len = (uint32_t)mono_file_map_size (map);
+		g_assert (arg->runtimeconfig.data.data_len > 0);
+		buffer = (char*)mono_file_map (arg->runtimeconfig.data.data_len, MONO_MMAP_READ|MONO_MMAP_PRIVATE, mono_file_map_fd (map), 0, &ret_handle);
+		g_assert (buffer);
+
+		arg->runtimeconfig.data.data = g_new0 (char, arg->runtimeconfig.data.data_len);
```

Comment:
Remember that `arg->runtimeconfig` is a union.  So only one half of it is active at a time.  when you write to `arg->runtimeconfig.data.data` you're overwriting `arg->runtimeconf.name.path`.

The problem is if a user gives you a cleanup function, they will expect that the `arg` you pass back to them still has a `name.path` that is the same as what they gave you (for example if they did a malloc to allocate the path string, they will want to free it in the cleanup function).  So you can't just overwrite what's passed in.

Also in the end I think if we keep pushing more and more work into `_install` we can avoid copying the data out of the buffer here.

Reply:
Make sense. Fixed. Thanks!